src/main/java/com/fasterxml/jackson/core/util/BufferRecyclers.java

```diff
@@ -180,4 +185,38 @@ public static void quoteAsJsonText(CharSequence input, StringBuilder output) {
     public static byte[] quoteAsJsonUTF8(String rawText) {
         return JsonStringEncoder.getInstance().quoteAsUTF8(rawText);
     }
+
+    /*
+    /**********************************************************************
+    /* Default BufferRecyclerProvider implementations
+    /**********************************************************************
+     */
+
+    public static BufferRecyclerProvider defaultProvider() {
+        return ThreadLocalBufferRecyclerProvider.INSTANCE;
+    }
+
+    /**
+     * Default {@link BufferRecyclerProvider} implementation that uses
+     * {@link ThreadLocal} for recycling instances.
+     *
+     * @since 2.16
+     */
+    public static class ThreadLocalBufferRecyclerProvider
```

Comment:
Was going back and forth between creating a main-level class vs nested here; mostly located here to keep close to `defaultProvider()` defined above.

Reply:
I agree that it's better to keep all default implementations in one place and have them internal to the main level class. By the way we will also need another implementation not pooling anything but simply returning a new `BufferRecycler` every time that `acquire` is called. I believe that this "dummy" implementation should be also placed here.

Reply:
Oh. The reason -- maybe not great one -- was that the check for "do not recycle" is using a simple Feature, and by-passes provider/pool altogether.

Maybe using and passing no-op instance would make more sense, nonetheless, instead of using `null` provider/pool.